opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -95,14 +95,103 @@ message InstrumentationLibraryMetrics {
+//
+// StartTimeUnixNano may be used with Gauge instruments to indicate
+// when the measurement was captured, as opposed to the moment it was
+// observed.
+//
+// Another use for this information is to prevent overlap when
+// multiple writers accidentally write to a single stream--applies
+// to all data points.
+//
+// # Unknown StartTimeUnixNano
+//
+// This protocol adopts the Prometheus heuristic for reset detection,
+// and with it prescribes a solution for filling in StartTimeUnixNanos
+// with as much information as possible to assist consumers with
+// calculating rates and detecting overlap.
+//
+// The Prometheus heuristic indicates that a reset must have occured
+// in a monotonic series whenever the value descends from one value to
+// the next, in a timestamp-ordered sequence.
+//
+// To facilitate reset and overlap detection, when single-timestamped
+// sample values are translated into OTLP, such as from the Prometheus
+// Remote-Write protocol, use the following.
+//
+// ## Filling in StartTimeUnixNano
+//
+// To import a sequence of non-overlapping measurements into into OTLP
+// using a stateful processor:
+//
+// 1. Keep a map, using the metric name and attributes as the key, containing
+//    a struct of three parts:
+//    - resetTimestamp: the first observation or the reset timestamp
+//    - previousTimestamp: the timestamp of the most recent observation
+//    - previousValue: a monotonic value
+// 2. When a new point arrives and it is the first observation:
+//    - in the map, set resetTimestamp to the input timestamp
+//    - in the map, set previousValue to the input value
+//    - in the map, set previousTimestamp to resetTimestamp
+//    - in the output, set StartTimeUnixNano to resetTimestamp
+// 3. For each subsequent point that arrives in the sequence:
+//    - if the sample value is less than previousValue,
+//      then set resetTimestamp to previousTimestamp
+//    - set previousValue to the input value
+//    - set previousTimestamp to the input timestamp
+//    - in the output, set StartTimeUnixNano to resetTimestamp
+//
+// This technique ensures that all output points have a meaningful
+// StartTimeUnixNano.  The first point the stream written by this
+// writer has TimeUnixNano equal to StartTimeUnixNano, a point with
+// zero duration.  Consumers of this data are expected to recognize
+// these zero-width points as stream resets where the true start time
+// is unknown.
+//
+// Using StartTimeUnixNano values filled in this way, consumers are
+// able to calculate rates correctly and easily, without
+// reconstructing a sequence of ordered points to apply the Prometheus
+// heuristic.  Receivers of cumulative and delta metric data with
+// unknown start time are strongly encouraged to apply this logic as
+// while translating into OTLP, to avoid producing data points with
+// AggregationTemporality and unknown StartTimeUnixNano.
+//
+// The map can be cleared periodically, in effect resetting streams.
```

Comment:
Nit: You could keep a heuristic notion of the metric report rate to judge when you're beyond N-sigma from last expected point timestamp to clear.

I think it's good to call out that this algorithm is *one* way to solve the problem, and could be enriched (similar to what we're trying to say in Delta->Cumulative algorithm).  I.e. we aren't forcing OTLP users to write exactly this, we're just outlining a "simple" approach to handling the scenario.

Reply:
I added
```
// Note: This is meant to provide guidance in importing data into OTLP
// that does not have start time and there is no other form of reset
// detection.
```
 at the top and
 ```
 // Enties can safely be removed from the map, after they appear no
// longer used.  This has the effect of entering a gap stream where
// the processor reset its memory of the start time.
```
at the bottom, how's that?